### DISCUSSION TITLE
安装lanms-neo, Polygon3报错

### DISCUSSION THREAD

**Discussion by gch9807** [PaddlePaddle/PaddleOCR#12617] [Q&A] [answered]
Building wheels for collected packages: lanms-neo, Polygon3
  Building wheel for lanms-neo (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for lanms-neo (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [18 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-39
      creating build\lib.win-amd64-cpython-39\lanms
      copying lanms\__init__.py -> build\lib.win-amd64-cpython-39\lanms
      running build_ext
      building 'lanms._C' extension
      creating build\temp.win-amd64-cpython-39
      creating build\temp.win-amd64-cpython-39\Release
      creating build\temp.win-amd64-cpython-39\Release\lanms
      creating build\temp.win-amd64-cpython-39\Release\lanms\csrc
      creating build\temp.win-amd64-cpython-39\Release\lanms\csrc\clipper
      "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.37.32822\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\gch\AppData\Local\Temp\pip-install-en6bm2ae\lanms-neo_f0823f63a45d420aaf4f822a4e64cb9d\lanms\csrc -IC:\Users\gch\AppData\Local\Temp\pip-build-env-bwk1kcg7\overlay\Lib\site-packages\pybind11\include -IC:\Users\gch\.conda\envs\paddle2\include -IC:\Users\gch\.conda\envs\paddle2\Include "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.37.32822\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include" /EHsc /Tplanms/csrc/clipper\clipper.cpp /Fobuild\temp.win-amd64-cpython-39\Release\lanms/csrc/clipper\clipper.obj /std:c++latest /EHsc /bigobj
      clipper.cpp
      C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.37.32822\include\cstdlib(12): fatal error C1083: 无法打开包括文件: “math.h ”: No such file or directory
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.37.32822\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for lanms-neo
  Building wheel for Polygon3 (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      Using NumPy extension!
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-39
      creating build\lib.win-amd64-cpython-39\Polygon
      copying Polygon\IO.py -> build\lib.win-amd64-cpython-39\Polygon
      copying Polygon\Shapes.py -> build\lib.win-amd64-cpython-39\Polygon
      copying Polygon\Utils.py -> build\lib.win-amd64-cpython-39\Polygon
      copying Polygon\__init__.py -> build\lib.win-amd64-cpython-39\Polygon
      running build_ext
      building 'Polygon.cPolygon' extension
      creating build\temp.win-amd64-cpython-39
      creating build\temp.win-amd64-cpython-39\Release
      creating build\temp.win-amd64-cpython-39\Release\src
      "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.37.32822\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -DDEFAULT_STYLE=STYLE_LIST -DWITH_NUMPY=1 -DSYSTEM_WIN32=1 -Isrc -IC:\Users\gch\.conda\envs\paddle2\lib\site-packages\numpy\core\include -IC:\Users\gch\.conda\envs\paddle2\include -IC:\Users\gch\.conda\envs\paddle2\Include "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.37.32822\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include" /Tcsrc/PolyUtil.c /Fobuild\temp.win-amd64-cpython-39\Release\src/PolyUtil.obj
      PolyUtil.c
      C:\Users\gch\AppData\Local\Temp\pip-install-en6bm2ae\polygon3_5cc6be714fc148bcbb680b4d3c2d3e24\src\gpc.h(36): fatal error C1083: 无法打开包括文件: “stdio.h”: No such file or directory
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.37.32822\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for Polygon3
  Running setup.py clean for Polygon3
Failed to build lanms-neo Polygon3
ERROR: Could not build wheels for lanms-neo, Polygon3, which is required to install pyproject.toml-based projects

**Comment by noiiiu** [comment]
i have solved this problem by install visual studio C++, i selected MSVC、windows 10/11 SDK、C++ CMake tool、generate tool and C++ AddressSanitizer

**Reply by noiiiu** [reply]
hopefully it may be helpful

**Reply by nadirvishun** [reply]
I am using Linux, but I still get this error.

**Reply by nadirvishun** [reply, marked as answer]
I'am useing `Alpine`,  Successfully run after installed the following dependencies: 
```sh
apk update && apk add --no-cache build-base python-3.11-dev
```
because my python is 3.11 so install `python-3.11-dev`, otherwise run install `python-dev` for latest version

if use `Ubuntu` can try (without actual testing, it's uncertain whether `python-dev` needs to be installed): 
```sh
apt-get update && apt-get install -y build-essential
```
